### PR TITLE
[openstack_designate] Add new plugin

### PR DESCRIPTION
**Pull request by bregman-arie**
This adds a plugin for OpenStack Designate.

Resolves: #2150

Signed-off-by: abregman <abregman@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [x] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

File: sos/report/plugins/openstack_designate.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            ])
+        else:
+            self.add_copy_spec([
+                "/var/log/designate/*.log",
+                "/var/log/containers/designate/*.log"
+            ])
+
+    def postproc(self):
+        protect_keys = [
+            "password", "connection", "transport_url"
```
Comment (johnsom) [inline sos/report/plugins/openstack_designate.py:43]: This is an incomplete list.

**Comment by TurboTurtle** [comment]
Just to cycle around on this - we'll be closing 4.0 in roughly 11 days. I can see from the referenced RHBZ there's a number of commands that need to be collected that aren't yet in the plugin.

I don't immediately see anything that needs changing with the current version of the plugin, so if those commands can be added this should be able to be included in the 4.0 cut.

**Comment by TurboTurtle** [comment]
@bregman-arie 4.0 will be closing on Monday the 17th now - do you want to update this plugin with the commands specified in the RHBZ, or do you want to take this as-is for the 4.0 cut and extend it at a later time?

**Comment by johnsom** [comment]
I am going to try to post a followup patch this afternoon that adds the rest of the BZ items. I think they are mostly the command output items left.

File: sos/report/plugins/openstack_designate.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            self.var_puppet_gen + "/etc/designate/*",
+            regexp, r"\1*********"
+        )
+
+
+class RedHatdesignate(OpenStackDesignate, RedHatPlugin):
+
+    packages = ('openstack-selinux',)
+
+# vim: set et ts=4 sw=4 :
```
Comment (slashdd) [inline sos/report/plugins/openstack_designate.py:60]: Could you please add an Ubuntu class as follows :

```
class Ubuntudesignate(OpenStackDesignate, UbuntuPlugin):

    packages = ('designate-common',)
```

File: sos/report/plugins/openstack_designate.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin
```
Comment (slashdd) [inline sos/report/plugins/openstack_designate.py:9]: Could you please add `UbuntuPlugin` ?

File: sos/report/plugins/openstack_designate.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        self.do_path_regex_sub("/etc/designate/*", regexp, r"\1*********")
+        self.do_path_regex_sub(
+            self.var_puppet_gen + "/etc/designate/*",
+            regexp, r"\1*********"
+        )
+
+
+class RedHatdesignate(OpenStackDesignate, RedHatPlugin):
+
+    packages = ('openstack-selinux',)
```
Comment (slashdd) [inline sos/report/plugins/openstack_designate.py:58]: I'm not familiar with designate installation on RHEL, but the package name caught my eye here. Is this the right package name needed ? 
https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/8/html/dns-as-a-service_guide/manual_dnsaas_installation

**Pull request closed without merging**